packages/migrator/src/cli.spec.ts

```diff
@@ -0,0 +1,271 @@
+import path from 'path'
+
+import execa from 'execa'
+import IORedis from 'ioredis'
+import { GenericContainer } from 'testcontainers'
+
+import type { Monorepo } from '@/utils/tests/repo'
+import type { ExecaReturnValue, Options as ExecaOptions } from 'execa'
+import type { Redis } from 'ioredis'
+import type { StartedTestContainer } from 'testcontainers'
+
+import { getAppKeyPrefix, getAppPrefixedKey } from '@/utils/kvdb/keyPrefix'
+import { generateAppKeys, fillKVDB, getAllKeys } from '@/utils/tests/kvdb'
+import { createTestMonoRepo } from '@/utils/tests/repo'
+
+function expectSuccessfulExecution (result: ExecaReturnValue) {
+    const { stdout, stderr, exitCode } = result
+    expect(exitCode).toEqual(0)
+    expect(stderr).toBe('')
+    expect(stdout).not.toBe('')
+}
+
+let currentDB = -1
```

Comment:
You can use `INFO keyspace` command to discover active databases.
That command will give you something like that:
```
db0:keys=10,expires=0
db1:keys=1,expires=0
db3:keys=1,expires=0
```

Btw your approach will work too)

Reply:
I guess your solution is more ultimate, but It has some cons: 
To extract connectionString via `info keyspace` you need connected client... And to connect client you need connection string 😅 

I can possibly connect to 0-indexed one, receive free index and re-connect, but I guess its overhead for now

But thanks for suggestion, will keep it in mind in case of future refactoring